app/workers/backend_delete_end_users_worker.rb

```diff
@@ -3,8 +3,15 @@
 class BackendDeleteEndUsersWorker
   include Sidekiq::Worker
 
+  sidekiq_options queue: :backend_sync
+
   def perform(event_id)
     event = EventStore::Repository.find_event!(event_id)
-    ThreeScale::Core::User.delete_all_for_service(event.service_id)
+    service_id = event.service_id
+    ThreeScale::Core::User.delete_all_for_service(service_id)
+  rescue ActiveRecord::RecordNotFound => exception
+    System::ErrorReporting.report_error(exception, parameters: {event_id: event_id})
+  rescue ThreeScale::Core::APIClient::APIError => exception
+    System::ErrorReporting.report_error(exception, parameters: {event_id: event_id, service_id: service_id})
```

Comment:
Can't you mix these two rescues? They are the same except for service_id.
Would that make sense?

Reply:
@josemigallas The `service_id` is not present for the other one and I definitely need it for this one 🤔

Reply:
Pushed, now they are in the same rescue and when service_id is not present, it will just send `nil` 😄

Reply:
Hmm. If it fails with `ActiveRecord::RecordNotFound`, then it has no `service_id` defined while rescuing and it will raise `NoMethodError`.

Reply:
It is just `nil` @guicassolato , I checked it before 😄 Look at this:
![image](https://user-images.githubusercontent.com/11318903/54210512-ea407780-44df-11e9-97f5-1e27254b852e.png)

Otherwise the tests would fail anyway 😉